modules/actions/snapshots/lib/fill-missing-timestamps.ts

```diff
@@ -0,0 +1,50 @@
+import { Prisma } from '@prisma/client';
+import { secondsPerDay } from '../../../common/time';
+import _ from 'lodash';
+
+export const fillMissingTimestamps = (
+    items: Prisma.PrismaPoolSnapshotUncheckedCreateInput[],
+): Prisma.PrismaPoolSnapshotUncheckedCreateInput[] => {
+    // Group items by poolId
+    const grouped = _.groupBy(items, 'poolId');
+
+    // Iterate over each poolId group
+    const filledGroups = _.flatMap(grouped, (groupItems, poolId) => {
+        // Sort the group by timestamp
+        const sortedItems = _.sortBy(groupItems, 'timestamp');
+
+        if (sortedItems.length === 0) {
+            return [];
+        }
+
+        // Find the range of timestamps
+        const timestamps = sortedItems.map((item) => item.timestamp);
+        const minTimestamp = _.min(timestamps) ?? 0;
+        const maxTimestamp = _.max(timestamps) ?? 0;
+
+        // Generate all timestamps within the range
+        const allTimestamps: number[] = [];
+        for (let ts = minTimestamp; ts <= maxTimestamp; ts += secondsPerDay) {
+            allTimestamps.push(ts);
+        }
+
+        // Fill missing timestamps
+        const filledItems: Prisma.PrismaPoolSnapshotUncheckedCreateInput[] = [];
+        let lastKnownItem: Prisma.PrismaPoolSnapshotUncheckedCreateInput | null = null;
+
+        for (const timestamp of allTimestamps) {
```

Comment:
here you could just compare the length of `allTimestamps` and `sortedItems` and if they match, add all.